dias/tracker.py

```diff
@@ -0,0 +1,418 @@
+                )
+                return None
+
+            if not db.table_exists("file"):
+                print("File Tracker: Creating file table")
+                db.create_tables([FileTable])
+
+                files = []
+                # grab a list of files on disk of FILETYPE
+                for ft in FILETYPES:
+                    files.extend(
+                        glob.glob(
+                            os.path.join(
+                                self.base_path,
+                                f"*_{ft}_corr",
+                                "*.h5",
+                            )
+                        )
+                    )
+
+                self.insert_files(files)
+
+            if not db.table_exists("processed"):
+                print("File Tracker: Creating processed table")
+                db.create_tables([Processed])
+
+    def _get_filetypes(self):
+        """
+        Get a dictionary filled with peewee rows referencing all of the filetypes in SQL db.
+
+        Returns
+        -------
+        dict
+            Key: String - name of filetype
+            Value: Model instance referencing row for filetype
+        """
+        return {ft: FileType.get(name=ft) for ft in FILETYPES}
+
+    def insert_files(self, files):
+        """
+        Insert files into File Table of SQL db. For them sets exists = True.
+
+        Files that already exist in db are ignored.
+
+        Parameters
+        ----------
+        files : List of String
+            list of filepaths
+        """
+        # pattern used to extract filetype of file from its name
+        pattern = re.compile(".*Z_(.*)_corr.*")
+
+        # represents each row in table
+        record = namedtuple(
+            "record", "filepath file_type_id, start_time end_time exists"
+        )
+
+        records = []  # list of `record`s
+
+        file_type_rows = self._get_filetypes()
+
+        print(f"File Tracker: Adding {len(files)} to database")
+
+        for f in files:
+            source = h5py.File(f, "r")
+            file_type = pattern.search(f).group(1)
+            records.append(
+                record(
+                    f,
+                    file_type_rows[file_type],
+                    source["index_map/time"][0]["ctime"],
+                    source["index_map/time"][-1]["ctime"],
+                    True,
+                )
+            )
+
+        # some versions of SQLIte accept only 100 records in a batch insert at a time
+        for batch in chunked(records, 100):
+            FileTable.insert_many(
+                batch,
+                fields=[
+                    FileTable.filepath,
+                    FileTable.file_type_id,
+                    FileTable.start_time,
+                    FileTable.end_time,
+                    FileTable.exists,
+                ],
+            ).on_conflict(action="IGNORE").execute()
+            # on_conflict specifies that we do not insert duplicates
```

Comment:
Is there any reason not to just insert them as you scrape the files? For a local sqlite db, it's not going to be very costly doing all the inserts, and the only time you'd have a lot of operations to do is on the first pass when there's no files in the db.